internal/docker/docker.go

```diff
@@ -54,9 +54,13 @@ func NewDockerClient(config *Config, logger *zap.Logger, opts ...docker.Opt) (*C
 			return nil, err
 		}
 	}
+	endpoint := docker.DefaultDockerHost
+	if config.Endpoint != "" {
+		endpoint = config.Endpoint
+	}
```

Comment:
Isn't this already covered by the default config? I mean when the default config is created the endpoint is correct and when you unmarshall the YAML it only overwrites if the user configured the endpoint.